### ISSUE TITLE
Can we please have GPU support?

### ISSUE THREAD

**Issue by xonaman** [rainyl/opencv_dart#219]
**Is your feature request related to a problem? Please describe.**
Not a problem, but GPU utilization would make a good performance boost.

**Describe the solution you'd like**
OpenCV offers types like cv::GpuMat via the CUDA module, which bindings could be implemented to.

**Describe alternatives you've considered**
GPU is _THE_ alternative to CPU image processing.

**Additional context**
The following graphic from [here](https://opencv.org/platforms/cuda/) should say all you need to know.
![perf](https://github.com/user-attachments/assets/e287c6cd-8f8e-4d64-afac-235ca4df4b94)

**Comment by rainyl** [comment]
Yes, it's in consideration. But it's just not easy and lots of works todo, and i have no much time, will work on in the future.

**Comment by abdelaziz-mahdy** [comment]
what needs to be implemented to make it work? maybe i can help in it when i get time too

**Comment by rainyl** [comment]
@abdelaziz-mahdy basically something like https://github.com/hybridgroup/gocv/tree/release/cuda

and of course, a nvidia GPU is required to test it🤣

**Comment by abdelaziz-mahdy** [comment]
> @abdelaziz-mahdy basically something like https://github.com/hybridgroup/gocv/tree/release/cuda
> 
> and of course, a nvidia GPU is required to test it🤣

Mission failed for me then 😂

**Comment by rainyl** [comment]
me too. I am developing on a laptop without a NVIDIA GPU equipped, so not active on this.

If any one want to work on this, refer to https://github.com/hybridgroup/gocv/tree/release/cuda

**Comment by frg-x** [comment]
> @abdelaziz-mahdy basically something like https://github.com/hybridgroup/gocv/tree/release/cuda
> 
> and of course, a nvidia GPU is required to test it🤣
Why do you recall CUDA if mobile devices have their own GPUs, not related to CUDA?

**Comment by abdelaziz-mahdy** [comment]
> Why do you recall CUDA if mobile devices have their own GPUs, not related to CUDA?
> 
> > @abdelaziz-mahdy basically something like https://github.com/hybridgroup/gocv/tree/release/cuda
> > 
> > and of course, a nvidia GPU is required to test it🤣
> 

I think he wants it for windows and Linux, 

Mobile gpu support can be achieved using vulkan, but it's performance is still not that good I think and I don't know if matgpu supports it or not

**Comment by rainyl** [comment]
Well, GPU backends such as OpenCL and Vulkan on mobile devices have been supported.

**Comment by frg-x** [comment]
How to enable GPU processing in? Is GPU-processing implemented in `opencv_dart` API?

> Well, GPU backends such as OpenCL and Vulkan on mobile devices have been supported.

**Comment by rainyl** [comment]
> How to enable GPU processing? Is GPU-processing support implemented in `opencv_dart` API?
> 
> > Well, GPU backends such as OpenCL and Vulkan on mobile devices have been supported.

What processing? if you mean the methods like "cv::cuda::threshold", they are not supported by opencv on mobile devices.

GPU backends like OpenCL are used for dnn-related inferences, they are supported both in opencv_core and opencv_dart for now. If you want to enable them, please refer to the document of opencv.
